docs/content/3.middleware/2.guides/6.headers.md

```diff
@@ -0,0 +1,113 @@
+# Response Headers
+
+## Overview
+
+The Middleware Response Headers [extension](/middleware/guides/extensions) enables the addition of custom response headers, which can be particularly useful for managing application caching or establishing security rules. 
+
+With this extension, you can overwrite or set headers for specific API methods or apply them globally across your application. 
+It can be implemented for any Unified Alokai Backend integration that exposes API methods and allows the addition of custom headers.
+
+## How it works?
+
+The extension can be used in two ways:
+
+- **Locally**: to set custom headers for specific API methods
+- **Globally**: to set `Cache-Control` header for all the API methods
+  
+Internally, the extension will identify all GET-type methods and add/overwrite the headers to their responses using the `afterCall` hook. Read the [Middleware Extensions guide](/middleware/guides/extensions#creating-an-extension) to learn more about the internals of an extension.
+
+## Installation
+
+To install the extension, run the following command in your `apps/storefront-middleware` folder:
+
+```bash
+yarn add @vsf-enterprise/middleware-headers
+```
+
+## Configuration
+
+To configure the extension for a specific integration, add the following code to your `middleware.config.js` file:
+
+```typescript [middleware.config.ts]
+import headersExtension from '@vsf-enterprise/middleware-headers';
+
+import { extensionConfig } from './extensionConfig';
+
+export default {
+  integrations: {
+    // ... other integrations
+    [integrationName]: {
+      extensions: (extensions: ApiClientExtension[]) => [
+        ...extensions,
+        headersExtension({ ...extensionConfig }),
+        // ... other extensions
+      ]
+    }
+  },
+};
+```
+
+## Usage
+
+### Local Usage
+
+Now, in the `extensionConfig.ts` file, you can set up your headers for individual API methods.
+
+```typescript [extensionConfig.ts]
+export const extensionConfig = {
+  methods: {
+    [apiMethodName]: {
+      response: {
+        headers: {
+          'Cache-Control': 'your-rules',
+          // ... other headers
+        }
+      }
+    }
+  }
+};
+```
+
+Here, your IDE will be able to suggest the available API methods and their names. You can also define your custom ones, making sure that the method name matches the one from the API.
```

Comment:
How will the IDE suggest API method names? Is there a certain type-cast that has to be done on the `extensionConfig` object?